templates/puppet_managed.epp

```diff
@@ -0,0 +1,20 @@
+<%- | String  $message = 'WARNING This file is managed by puppet. Do not edit!',
+      String  $begin_line = '#',
+      String  $end_line = '',
+      Int     $line_length = 70,
+      String  $metadata_title = 'Metadata:',
+      Hash    $metadata = {}
+| -%>
```

Comment:
This is complex and too many tags to understand.  Would prefer to have the logic in the function.